Fronted features and fixes

mds1 opened the pull request:
Changes in this PR:

1. Lets user choose gas price when withdrawing ETH: The default gas price to use is queried from the node, and if the user wants to change it, they can. See #202 and #203 for more info / video demos

2. Adds payment links: On the send form, fill in any combination of fields and click the new "Generate Payment Link" button. This puts a URL on the clipboard that, when visited, pre-populates the form with the info in the link. It should work even if you visit the send page without a wallet connected

3. Prevents the "Unknown account #0" error when user rejects wallet connect prompt (resolves #192). More info in #205 

4. Fixes small UI error where modal is not on top of the page. More info in #205 

Thanks to @davidlaprade for helping with most of these!

apbendi reviewed (commented):
Overall looks and works great. Nice job @mds1 and @davidlaprade 🎉 

I'm seeing a bunch of test failures of `umbra-js` on this branch. Can you reproduce that? Those tests do not fail on master for me.

Here are some things I noticed in my testing:

* When the "Confirm Withdrawal" screen first pops up, it does a weird resizing thing where it shows up big then shrinks down:

![umbra-withdrawal-flash](https://user-images.githubusercontent.com/1479802/127591440-e8f707e9-610b-479b-87d8-c6ea5a4135e8.gif)

* Is there an underscore after this edit icon?

<img width="182" alt="Screen Shot 2021-07-29 at 10 37 58 PM" src="https://user-images.githubusercontent.com/1479802/127591560-d0058aee-cd2d-4a9a-89cf-2bc90c9827f4.png">

* Instead of "Generate payment link", how about "Copy Shareable Payment Link" with the little clipboard icon for the copy?
* After I click confirm, the gas price I've selected is still editable, when it really shouldn't be:

<img width="440" alt="Screen Shot 2021-07-29 at 10 42 35 PM" src="https://user-images.githubusercontent.com/1479802/127591963-d348e4a4-be42-45c5-8086-bf96da31ae58.png">

mds1 commented:
@apbendi Addressed all changes. Changed the button text to "Copy payment link" to keep a bit more concise, and added an FAQ question explaining the button

mds1 commented:
> I'm seeing a bunch of test failures of umbra-js on this branch. Can you reproduce that? Those tests do not fail on master for me.

I just ran `yarn` at the root and `yarn test` in the umbra-js folder and everything passed, so I'm not able to reproduce this. Can you try again and post the specific failures you're seeing?

apbendi commented:
Overall the changes look good! Great idea on the FAQ item. Copy looks good and good call out on the privacy implications.

One thing I noticed while testing— I'm not sure if it changed or if I missed it the first time— is that the fee does not "stick" when closing the form to edit it. When re-opening the form, the custom fee is retained, but when closing it, it reverts:

![Umbra-100-Gwei-Flip-Flop-Submission](https://user-images.githubusercontent.com/1479802/128261894-3e97251b-b603-49cd-82b4-96b59226956c.gif)

mds1 commented:
> One thing I noticed while testing— I'm not sure if it changed or if I missed it the first time— is that the fee does not "stick" when closing the form to edit it. When re-opening the form, the custom fee is retained, but when closing it, it reverts:

Ah, this was intentional. @davidlaprade and I thought that this approach made sense to handle the use case where a user might change their mind about a custom fee and just revert back to the default fee. If we wanted the fee to "stick" after clicking the X, then you'd either need a "reset" button or they'd have to close and re-open the modal to do so.

A use case for resetting is you accidentally enter a value that's too large or too small, and want to reset it because you don't know how check gas prices / or it's faster to do it this way / or you trust the default more than yourself

apbendi commented:
> Ah, this was intentional. @davidlaprade and I thought that this approach made sense to handle the use case where a user might change their mind about a custom fee and just revert back to the default fee. If we wanted the fee to "stick" after clicking the X, then you'd either need a "reset" button or they'd have to close and re-open the modal to do so.

> A use case for resetting is you accidentally enter a value that's too large or too small, and want to reset it because you don't know how check gas prices / or it's faster to do it this way / or you trust the default more than yourself

That makes sense, but was pretty un-intuitive, at least for my cave man brain. I clicked the "x" expecting to have that fee reflected. Perhaps we can make it more explicit that closing the edit form will throw away the custom fee— for example, what if we change the "x" to a button that said "Cancel"?
